util/chplenv/chpl_gmp.py

```diff
@@ -49,6 +60,17 @@ def get_compile_args():
     gmp_val = get()
     if gmp_val == 'bundled':
          return third_party_utils.get_bundled_compile_args('gmp')
+    elif gmp_val == 'system':
```

Comment:
Just a thought, but this idiom of checking pkg-config and then homebrew has shown up in several of your PRs I've reviewed lately. Maybe it should be a function?

Reply:
I can do that. I was also discussing this PR with @mppf yesterday and he suggested that the `brew` checking was fragile and not worth the maintenance effort. So instead the logic would be
```
if pkg_config:
  use pkg_config
elif brew and pkg exists in brew (and no pkg_config):
  error, tell user to `brew install pkg-config`
else:
  error, tell user to install pkg or not to use system
```

That way we don't have to guess the flags. I can add this logic to this PR and update jemalloc and hwloc as well